cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py

```diff
@@ -990,24 +988,55 @@ def ceph_storage_pool():
     :return: None
     '''
     hookenv.log('Creating Ceph pools.')
+    # note that we do not delete pools when
+    # swapping configurations on purpose to preserve
+    # any data a user may have
+    if ceph_using_csi():
+
+        ReplicatedPool(
+            name='xfs-pool',
+            service='admin',
+            replicas=3,
+            app_name=None,
+        ).create()
+
+        ReplicatedPool(
+            name='ext4-pool',
+            service='admin',
+            replicas=3,
+            app_name=None,
+        ).create()
+    else:
 
-    ReplicatedPool(
-        name='xfs-pool',
-        service='admin',
-        replicas=3,
-        app_name=None,
-    ).create()
-
-    ReplicatedPool(
-        name='ext4-pool',
-        service='admin',
-        replicas=3,
-        app_name=None,
-    ).create()
+        ReplicatedPool(
+            name='rbd',
+            service='admin',
+            replicas=3,
+            app_name=None,
+        ).create()
```

Comment:
I think ceph creates an `rbd` pool by default. The charm never created it before, right? I don't think users had to either.

Reply:
What happens when the pools already exist? Does `ReplicatedPool(...).create()` raise an exception?

I ask because, for the first time, we now have code that clears the`kubernetes-master.ceph.pool.created` flag. This handler could run multiple times.

Reply:
Ok, so here's the deal. If there is no pool ceph will create an rdb pool the first time you request something. If you started out with CSI though it creates two pools and ceph won't create the rdb pool. That is why I added it.

Also, no exception as I switched between the two multiple times without issue.

Reply:
Ooh that makes sense. Thanks!